### PR TITLE
Adding Convenience Method to Configurator interface / ServiceConfigGuide class

### PR DESCRIPTION
**Pull request by nixon124**

### CODE REVIEW

File: src/main/java/emissary/config/ServiceConfigGuide.java [hunk 1]
```diff
@@ -845,6 +846,20 @@ public Map<String, Set<String>> findStringMatchMultiMap(@Nullable final String p
...
 
+    /**
+     * Return the first string entry matching the key parameter
+     *
+     * @param theParameter key to match
+     * @return the first matching value
+     * @throws IllegalArgumentException if no non-blank value is found
+     */
+    @Override
+    public String findRequiredStringEntry(final String theParameter) throws IllegalArgumentException {
```
Comment (drivenflywheel) [inline src/main/java/emissary/config/ServiceConfigGuide.java:857]: Please add unit test coverage that includes one or more valid entries, a zero-length entry (`FOO = ""`), and a missing entry

Comment (nixon124) [reply src/main/java/emissary/config/ServiceConfigGuide.java:857]: Got it.

File: src/test/java/emissary/config/ServiceConfigGuideTest.java [hunk 2]
```diff
@@ -611,4 +611,31 @@ void testFindStringEntry_WithDefaultAndConfiguredNull_Emissary201() {
...
+        config.addEntry(emptyProp, emptyValue);
+
+        // Working entry
+        final String value = config.findRequiredStringEntry(workingProp);
+        assertEquals(value, nonNullValue, "Does this work?");
+
+        // Zero-Length Entry
+        assertThrows(IllegalArgumentException.class, () -> {
+            config.findRequiredStringEntry(emptyProp);
+        });
```
Comment (drivenflywheel) [inline src/test/java/emissary/config/ServiceConfigGuideTest.java:634]: For future reference... The `assertThrows` call actually returns the thrown Exception, so you could capture it and inspect its message.
```suggestion
        Exception ex = assertThrows(IllegalArgumentException.class, () -> config.findRequiredStringEntry(emptyProp));
        assertNotNull(ex.getMessage(), "exception message should not be null");
        assertTrue(ex.getMessage().contains(emptyProp), "Exception message should name the offending config entry");
        
```

**Review by drivenflywheel — APPROVED** [review]

**Review by drivenflywheel — APPROVED** [review]
🏅

**Comment by cfkoehler** [comment]
@nixon124 pipeline and local build shows that the formatter needs to be run and those changes committed.

**Review by drivenflywheel — APPROVED** [review]

**Review by jpdahlke — APPROVED** [review]
As a follow-up it would be nice to use this somewhere to illustrate
